patient-appointment-management/functions/login.js

```diff
@@ -1,17 +1,40 @@
-/* eslint-disable prefer-destructuring, dot-notation */
+/* eslint-disable prefer-destructuring, dot-notation, prefer-template */
 exports.handler = function (context, event, callback) {
   const path = Runtime.getFunctions()['auth'].path;
-  const { createToken, isAllowed } = require(path);
-  const ac = context.ACCOUNT_SID;
+  const { isValidPassword, createMfaToken } = require(path);
 
-  const token = createToken(event.password, context);
   const response = new Twilio.Response();
   response.appendHeader('Content-Type', 'application/json');
 
-  // Short-circuits
-  if (isAllowed(token, context)) {
-    response.setBody({ token });
-    callback(null, response);
+  if (isValidPassword(event.password, context)) {
+    // Generate and send six digit MFA code to administrator's phone
+    const twilioClient = context.getTwilioClient();
+    mfaCode = Math.floor(100000 + Math.random() * 900000);
```

Comment:
If we use MFA we should rely on something like [Twilio Verify](https://www.twilio.com/verify). 2FA is a well defined algorithm and not just generating a random value.

Reply:
@dkundel , didn't know about Twilio Verify. we'll look into it.

Reply:
Dominik, changing to Twilio Verify appears to involve some effort, can we add this to v3 backlog?

Reply:
@bochoi-twlo you'll need to add a [Verify Service](https://twilio.com/console/verify/services), but otherwise it's changing the below to:

```
twilioClient.verify.services('VAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
             .verifications
             .create({
        to: context.ADMINISTRATOR_PHONE,
        channel: 'sms'
      })
```

This actually saves you a lot of code because you don't need to store the MFA token, the API will track it for you. I understand this changes a bit of your architecture but it's a much better way to send MFA tokens. See: https://www.twilio.com/blog/migrate-programmable-sms-to-verify